tidy3d/updater.py

```diff
@@ -202,6 +202,46 @@ def iterate_update_dict(update_dict: Dict, update_types: Dict[str, Callable]):
             iterate_update_dict(item, update_types)
 
 
+@updates_from_version("1.7")
+def update_1_7(sim_dict: dict) -> dict:
+    """Updates version 1.7."""
+
+    def fix_abstract_info(mnt_dict: dict) -> dict:
+        mnt_dict.pop("fields")
+        return mnt_dict
+
+    def fix_angle_info(mnt_dict: dict) -> dict:
+        mnt_dict.pop("fields")
+        mnt_dict["proj_distance"] = 1e6
+        return mnt_dict
+
+    def fix_cartesian_info(mnt_dict: dict) -> dict:
+        mnt_dict.pop("fields")
+        dist = mnt_dict.pop("plane_distance")
+        mnt_dict["proj_distance"] = dist
+        axis = mnt_dict.pop("plane_axis")
+        mnt_dict["proj_axis"] = axis
+        return mnt_dict
+
+    def fix_kspace_info(mnt_dict: dict) -> dict:
+        mnt_dict.pop("fields")
+        mnt_dict["proj_distance"] = 1e6
+        axis = mnt_dict.pop("u_axis")
+        mnt_dict["proj_axis"] = axis
+        return mnt_dict
+
+    iterate_update_dict(
+        update_dict=sim_dict,
+        update_types={
+            "AbstractNear2FarMonitor": fix_abstract_info,
```

Comment:
Is this ever used? There should never be an abstract monitor type in the json, right?

Reply:
Oh true, thanks.